### PR TITLE
New print.css

### LINKED ISSUE #223: Print CSS styles
Check that print css still works following redesign, fix if necessary

### PR DESCRIPTION
**Pull request by wrightmartin**
fixes #223

Pages should now hide all necessary elements when printed
No width restrictions
Very readable

### CODE REVIEW

**Comment by wrightmartin** [comment]
Weirdly I had to reboot my server before the changes would show up.

Thinking about it, we should leave this till after the colours roll out, as there will be tweaks-a-plenty


### Before
screen shot 2015-06-22 at 12 17 10

### After
screen shot 2015-06-22 at 12 28 12

**Comment by garethrees** [comment]
The WhatDoTheyKnow title has a scrollbar:

screen shot 2015-06-22 at 13 08 02

The print view should include the full URL of the request if possible. (It does on Mac if you check the "Headers and footers" box, but don't know about other systems?)

I prefer that the previous version used the sans-serif font – looks neater.

**Comment by wrightmartin** [comment]
I've pushed a few changes to this branch now, here's what it looks like

screen shot 2015-06-24 at 14 34 10

**Comment by garethrees** [comment]
```irc
[09:56:00]  <gareth>	 Martin just sent you two PDFs. I think the older one is much eaiser to follow in terms of who's saying what
[09:56:11]  <Martin>	 cheers gareth,will take a look
[09:57:46]  <Martin>	 yeah you're right, it needs to take up less space, but the sections make it easier
```

**Comment by wrightmartin** [comment]
Have made some revisions based on feedback

[new-print-styles.pdf](https://github.com/mysociety/whatdotheyknow-theme/files/506/new-print-styles.pdf)

**Comment by garethrees** [comment]
Looking good!

When I view in device emulation in chrome I still get colour, whereas "Print" is greyscale. What would actually happen if I printed it?

I don't particularly like the way the rounded borders look – seems to be nastier when rendering to PDF at least. I'll leave that up to you though.

File: assets/stylesheets/responsive/_print_style.scss [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        color: black;
+    }
+}
+
+.footer__about {
+    width: 100%;
+}
+
+//Help pages
+.controller_help {
```
Comment (garethrees) [inline assets/stylesheets/responsive/_print_style.scss]: `.controller_help` is also used on L73

File: assets/stylesheets/responsive/_print_style.scss [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+}
+
+.attachments {
+    margin-bottom: 1.5em;
+    img {
+        display: none;
+    }
+}
+
+.incoming.correspondence {
```
Comment (garethrees) [inline assets/stylesheets/responsive/_print_style.scss]: `.incoming.correspondence` is already defined on L161

File: assets/stylesheets/responsive/_print_style.scss [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    .button-secondary {
+        display: none;
+    }
+}
+
+.bodies-list strong a {
+    display: none;
+}
+
+// user profile
```
Comment (garethrees) [inline assets/stylesheets/responsive/_print_style.scss]: Delete if we're not going to style it

File: assets/stylesheets/responsive/_print_style.scss [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+p.subtitle {
+    margin-bottom: 0.5em;
+}
+
+.request_icon_line {
+    padding-left: 0;
+    background-image: none !important;
+}
+
+.correspondence {
```
Comment (garethrees) [inline assets/stylesheets/responsive/_print_style.scss]: `.correspondence` is already used on L73. This looks like the right place for it.

**Comment by garethrees** [comment]
Looks great – just some minor tidy-up. :+1:

File: assets/stylesheets/responsive/_print_style.scss (hunk 1 above)
Comment (wrightmartin) [inline assets/stylesheets/responsive/_print_style.scss]: That's part of a multiple declaration, not sure what I can do here?
```
.controller_help,
.correspondence {
```

Comment (garethrees) [inline assets/stylesheets/responsive/_print_style.scss]: Could extract the two component bits out in to mixins (or whichever is the one you call `extend` on!)

```scss
.expand_links {
  // ...
}

controller_help {
  @extend .expand_links;
}

.correspondence {
  @extend .expand_links;
}
```

I don't know which approach is best tbh – it was more of a thought.

Comment (wrightmartin) [inline assets/stylesheets/responsive/_print_style.scss]: Nope that's great, stole it wholesale :)

**Pull request merged**